tests/phpunit/tests/webfonts-api/providers/wpWebfontsGoogleProvider.php

```diff
@@ -0,0 +1,161 @@
+		$property = new ReflectionProperty( $this->provider, 'webfonts' );
+		$property->setAccessible( true );
+		$property->setValue( $this->provider, $webfonts );
+
+		$method = new ReflectionMethod( $this->provider, 'build_collection_api_urls' );
+		$method->setAccessible( true );
+		$actual = $method->invoke( $this->provider );
+
+		$this->assertSame( $expected, $actual );
+	}
+
+	/**
+	 * Data Provider.
+	 *
+	 * @return array
+	 */
+	public function data_build_collection_api_urls() {
+		return array(
+			'single font-family + single variation'    => array(
+				'webfonts' => array(
+					'open-sans.normal.400' => array(
+						'provider'     => 'google',
+						'font-family'  => 'Open Sans',
+						'font-style'   => 'normal',
+						'font-weight'  => '400',
+						'font-display' => 'fallback',
+						'is-external'  => true,
+					),
+				),
+				'expected' => array(
+					'https://fonts.googleapis.com/css2?family=Open+Sans:wght@400&display=fallback',
+				),
+			),
+			'single font-family + multiple variations' => array(
+				'webfonts' => array(
+					'open-sans.normal.400' => array(
+						'provider'     => 'google',
+						'font-family'  => 'Open Sans',
+						'font-style'   => 'normal',
+						'font-weight'  => '400',
+						'font-display' => 'fallback',
+						'is-external'  => true,
+					),
+					'open-sans.italic.700' => array(
+						'provider'     => 'google',
+						'font-family'  => 'Open Sans',
+						'font-style'   => 'italic',
+						'font-weight'  => '700',
+						'font-display' => 'fallback',
+						'is-external'  => true,
+					),
+				),
+				'expected' => array(
+					'https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,400;1,700&display=fallback',
+				),
+			),
+			'multiple font-families and variations'    => array(
+				'webfonts' => array(
+					'open-sans.normal.400' => array(
+						'provider'     => 'google',
+						'font-family'  => 'Open Sans',
+						'font-style'   => 'normal',
+						'font-weight'  => '400',
+						'font-display' => 'fallback',
+						'is-external'  => true,
+					),
+					'open-sans.italic.700' => array(
+						'provider'     => 'google',
+						'font-family'  => 'Open Sans',
+						'font-style'   => 'italic',
+						'font-weight'  => '700',
+						'font-display' => 'fallback',
+						'is-external'  => true,
+					),
+					'roboto.normal.900'    => array(
+						'provider'     => 'google',
+						'font-family'  => 'Roboto',
+						'font-style'   => 'normal',
+						'font-weight'  => '900',
+						'font-display' => 'fallback',
+						'is-external'  => true,
+					),
+				),
+				'expected' => array(
+					'https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,400;1,700&family=Roboto:wght@900&display=fallback',
+				),
+			),
+		);
+	}
+
+	private function get_webfonts_property() {
+		$property = new ReflectionProperty( $this->provider, 'webfonts' );
+		$property->setAccessible( true );
+
+		return $property->getValue( $this->provider );
```

Comment:
This method appears to be implemented differently in other test classes.

In `tests/phpunit/tests/webfonts-api/providers/wpWebfontsLocalProvider.php#L147`:
```php
private function get_webfonts_property() {
	$property = new ReflectionProperty( $this->provider, 'webfonts' );
	$property->setAccessible( true );

	return $property;
}
```

Then, in the same file at line 81:

```php
$this->assertSame( $webfonts, $property->getValue( $this->provider ) );
```

Reply:
The reason why they're different is to isolate the `test_get_css` test by directly setting the webfonts array into the property rather than using `set_webfonts()`. That said, I can see how this could be confusing when working on each test.

Good catch @costdev! I made them consistent in commit https://github.com/WordPress/wordpress-develop/pull/1736/commits/ed2499c80cf28433970edc47aa337cd5b209f4e4.